src/map/instance.cpp

```diff
@@ -809,352 +827,415 @@ enum e_instance_enter instance_enter(struct map_session_data *sd, unsigned short
 	return IE_OK;
 }
 
-/*==========================================
+/**
  * Request some info about the instance
- *------------------------------------------*/
-int instance_reqinfo(struct map_session_data *sd, unsigned short instance_id)
+ * @param sd: Player to display info to
+ * @param instance_id: Instance to request
+ * @return True on success or false on failure
+ */
+bool instance_reqinfo(struct map_session_data *sd, uint16 instance_id)
 {
-	struct instance_data *im;
-
-	nullpo_retr(1, sd);
+	nullpo_retr(false, sd);
 
 	if(instance_id == 0 || instance_id > MAX_INSTANCE_DATA)
```

Comment:
I wonder why those hard cap aren't into instance_search(), wouldn't this be more simple ?
From my understanding if instance_id==0 || instance_id > MAX_INSTANCE the search should fail no ?.
Or maybe the 1st id is 0.. just wondering